src/components/topthree.tsx

```diff
@@ -0,0 +1,155 @@
+//top3breed
+//separate component:
+//title, flexbox with 3 dogs
+import { useState, useEffect } from "react";
+import { readNameFormatter } from "../utils/readNameFormatter";
+import nameToImageURL from "../utils/nameToImageURL";
+import axios from "axios";
+
+interface TopDogsInterface {
+  breed_name: string;
+  count: number;
+}
+
+interface TopThreeProps {
+  top10Dogs: TopDogsInterface[];
+  toggle: React.Dispatch<React.SetStateAction<boolean>>;
+  toggleValue: boolean;
+}
+
+export default function TopThree(props: TopThreeProps): JSX.Element {
+  const top3Dogs = props.top10Dogs.slice(0, 3);
+
+  function handleClick() {
+    props.toggle(!props.toggleValue);
+  }
+  // dogImage variable contains the image file from our dogAPI
+  const [dogImage1, setDogImage1] = useState<string>("");
+  const [dogImage2, setDogImage2] = useState<string>("");
+  const [dogImage3, setDogImage3] = useState<string>("");
+
+  const voteMax = top3Dogs[0].count;
+  const dog1Score = 10;
+  // Stops the dogScore from exceeding 10
+  const dog2Score = Math.round((top3Dogs[1].count / voteMax) * 10);
+  const dog3Score = Math.round((top3Dogs[2].count / voteMax) * 10);
+
+  // fetching a random image of the topDogs based on breed_name
+  async function fetchImage1() {
+    const name_url = nameToImageURL(top3Dogs[0].breed_name);
+    const URL_STRING =
+      "https://dog.ceo/api/breed/" + name_url + "/images/random";
+    const response = await axios.get(URL_STRING);
+    setDogImage1(response.data.message);
+  }
+
+  async function fetchImage2() {
+    const name_url = nameToImageURL(top3Dogs[1].breed_name);
+    const URL_STRING =
+      "https://dog.ceo/api/breed/" + name_url + "/images/random";
+    const response = await axios.get(URL_STRING);
+    setDogImage2(response.data.message);
+  }
+
+  async function fetchImage3() {
+    const name_url = nameToImageURL(top3Dogs[2].breed_name);
+    const URL_STRING =
+      "https://dog.ceo/api/breed/" + name_url + "/images/random";
+    const response = await axios.get(URL_STRING);
+    setDogImage3(response.data.message);
+  }
+
+  //
```

Comment:
Nice job splitting up the functionality. Very clean and easy to read. Good structure